src/Core/Framework/Api/OAuth/ClientRepository.php

```diff
@@ -65,7 +65,12 @@ public function getClientEntity(string $clientIdentifier): ?ClientEntityInterfac
             return null;
         }
 
-        return new ApiClient($clientIdentifier, true, $values['label'] ?? Uuid::fromBytesToHex((string) $values['user_id']));
+        return new ApiClient(
+            $clientIdentifier,
+            true,
+            $values['label'] ?? Uuid::fromBytesToHex((string) $values['user_id']),
```

Comment:
if you also use the named parameter here, it would already be forward compatible. maybe also worth a hint in the changelog

Reply:
good idea 👍

Reply:
Although we do state here that named params are [not public API ](https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/6.5-new-php-language-features.md#backwards-compatibility--migration-strategy-3) 🙈

Reply:
that are our own coding guidelines :thinking: not sure, if that counts as ADR like thing set in stone :grimacing:

Reply:
Might be a good idea to document that somewhere, e.g. here: https://developer.shopware.com/docs/resources/guidelines/code/public-apis.html

that argument names (and consequently named arguments) are either part of our public API or not.

Also previously we had an extensive list with examples on what counts as a breaking change in our internal (old) handbook. I currently can't find that information anywhere 😬 . The documentation linked above only talks about some basics in PHP classes / services without going into details, but there is so much more: e.g. adding arguments to public methods in PHP services, storefront templates, vue components, ...

Reply:
wow, that is outdated :see_no_evil: and not even considering the feature of named arguments :grimacing:

Reply:
It's definitely a good idea to have that properly documenting but I think there needs to be a discussion and I don't have the capacity to shape that right now. I'm happy to support if you want to push it though :)